core/src/test/java/org/apache/iceberg/TestManifestReader.java

```diff
@@ -103,4 +107,28 @@ public void testManifestReaderWithUpdatedPartitionMetadataForV1Table() throws IO
     }
   }
 
+  @Test
+  public void testManifestReaderWithPartitionMetadataEvolution() throws IOException {
+    table.updateSpec().clear()
```

Comment:
I don't think these test cases should use `clear`. Most evolution should happen by adding and removing fields, so our tests should do the same.